sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -529,18 +530,32 @@ def set_read_only(
             404: ResourceNotFoundError
         }
 
+        match_condition = kwargs.pop("match_condition", MatchConditions.Unconditionally)
+        if match_condition == MatchConditions.IfNotModified:
+            error_map[412] = ResourceModifiedError
+        if match_condition == MatchConditions.IfModified:
+            error_map[412] = ResourceNotModifiedError
+        if match_condition == MatchConditions.IfPresent:
+            error_map[412] = ResourceNotFoundError
+        if match_condition == MatchConditions.IfMissing:
+            error_map[412] = ResourceExistsError
```

Comment:
quick clarifying question: if match condition is unconditionally, error_map at 412 doesn't have any default error?

Reply:
No. If it is unconditional, we have no idea why it returns 412. Hence we will raise HttpResponseError.